[MASTRA-2070] Update Vector Store functions to use object params

NikAiyer opened the pull request:
This PR converts createIndex, upsert and query to use object params for all current vector stores.

NikAiyer committed:
updated mastra vector and libsql to use object params

NikAiyer committed:
updated astra

NikAiyer committed:
update chroma

NikAiyer committed:
update pg

NikAiyer committed:
update pinecone

NikAiyer committed:
merge

NikAiyer committed:
update qdrant

NikAiyer committed:
update upstash

NikAiyer committed:
update vectorize

NikAiyer committed:
updated docs and examples

NikAiyer committed:
added changeset

NikAiyer committed:
updated changeset

NikAiyer committed:
fixed qdrant createIndex

NikAiyer committed:
update README

NikAiyer committed:
merge

NikAiyer committed:
update translate

NikAiyer committed:
Merge branch 'main' into mastra-2070

NikAiyer commented:
The current combined store test and RAG test errors seem unrelated to my changes (they were there before this PR), but looking into getting it fixed to make sure my test changes pass in CI

NikAiyer committed:
change timeout

NikAiyer committed:
Merge branch 'main' into mastra-2070

wardpeet commented on packages/core/src/vector/libsql/filter.ts:
```diff
@@ -1,5 +1,6 @@
 import { BaseFilterTranslator } from '../../filter';
 import type { FieldCondition, Filter, OperatorSupport } from '../../filter';
+import type { VectorFilter } from '../index.warning';
```
```suggestion
import type { VectorFilter } from '../index';
```

wardpeet reviewed (changes requested):
types should not live in index.ts files as you import them in other files as well which can lead to cyclical dependencies.

Move them to a types file and re-export them in index

NikAiyer committed:
Merge branch 'main' into mastra-2070

NikAiyer committed:
update exports for vector

NikAiyer committed:
Merge branch 'main' into mastra-2070

wardpeet commented on packages/core/src/vector/index.warning.ts:
```diff
@@ -1,4 +1,4 @@
-import { MastraVector as BaseMastraVector } from './index';
+import { MastraVector as BaseMastraVector } from './vector';
```
i think we do not need this anymore

NikAiyer replied:
Like we don't need the warning.ts anymore for vector?

NikAiyer committed:
update astra test

NikAiyer committed:
Merge branch 'main' into mastra-2070

NikAiyer committed:
Added normalize args for vector store operations (#2625)

This PR will be added to mastra-2070. It allows users to still use
positional args, but throws a deprecated warning when they are used.

NikAiyer committed:
Merge branch 'main' into mastra-2070

NikAiyer committed:
update chroma test

NikAiyer committed:
merge

NikAiyer committed:
update deprecation test for vectorize

NikAiyer committed:
another test fix

NikAiyer committed:
Merge branch 'main' into mastra-2070

TylerBarnes reviewed (approved):
🎉 huge improvement!!
